SpatialGDK/Source/SpatialGDKEditor/Private/SchemaGenerator/SpatialGDKEditorSchemaGenerator.cpp

```diff
@@ -1551,7 +1556,8 @@ bool RunSchemaCompiler(FString& SchemaBundleJsonOutput, FString SchemaInputDir,
 	}
 }
 
-bool ExtractComponentSetsFromSchemaJson(const FString& SchemaJsonPath, TMap<uint32, FComponentIDs>& OutComponentSetMap)
+bool ExtractInformationFromSchemaJson(const FString& SchemaJsonPath, TMap<uint32, FComponentIDs>& OutComponentSetMap,
```

Comment:
If we want to improve readability of the function this is in I think we should be factoring that code out into a couple of functions that would make it much clearer rather than having one monolithic thing and adding in macros to save space.

It looks like it has a couple of different stages e.g. opening the file, then iterating over different bits of it looking for different information. As it now has several concerns this might make it a bit easier to follow.

Reply:
Agree a bit, but I think this would be a larger split.
Basically the future version will probably be `ParseSchemaBundle(Path, SchemaDatabase&)` which iterates over almost everything in the json, using e.g. `ParseTypesInSchemaBundle(Fields, SchemaDatabase&)`, `ParseComponentsInSchemaBundle(Fields, SchemaDatabase&)`, `ParseComonentSetsInSchemaBundle(Fields, SchemaDatabase&)`. I think it will be easier to split according to the structure of the JSON to retain e.g. the scope we are in.
However, felt that the function was managable currently, could be debated.